src/app/components/post-modal/post-modal.component.ts

```diff
@@ -228,33 +209,35 @@ export class PostModalComponent {
       boardID: this.data.board.boardID,
       author: this.data.user.username
     }
-    this.commentService.add(comment).then(() => {
-      this.newComment = ''
-      this.comments.push(comment)
-    }).catch((e) => console.log(e))
+    this.canvasService.createCommentClient(comment, this.comments);
+    await this.canvasService.createCommentServer(comment);
+    this.newComment = '';
   }
 
-  handleLikeClick() {
+  async handleLikeClick() {
     // if liking is locked just return (do nothing)
     if(this.user.role == Role.STUDENT && !this.data.board.permissions.allowStudentLiking){
       return;
     }
       
     if (this.isLiked) {
-      this.likesService.remove(this.isLiked.likeID).then(() => {
-        this.isLiked = null
-        this.likes = this.likes.filter(like => like.likerID != this.user.id)
-      })
+      const postId = this.isLiked.postID;
+      const likeId = this.isLiked.likeID;
+
+      [this.likes, this.isLiked] = await this.canvasService
```

Comment:
@Ammar-T To continue the previous discussion, I've also found that some parts (such as this one) require the objects to be returned to the caller (otherwise it would cause a bug, but maybe there is an alternative solution?). If we were to combine into one function, that means the client update would be done only after the call to the firebase is completed. However, based on the original code, I don't think that's an intended behaviour

Reply:
Based on the original code, the client update only happens after the firebase call finishes. Look at the `.then(....`